File: src/main/java/com/github/novicezk/midjourney/dto/SubmitChangeDTO.java
package com.github.novicezk.midjourney.dto;

import com.github.novicezk.midjourney.enums.TaskAction;
import lombok.Data;
import lombok.EqualsAndHashCode;


@Data
@EqualsAndHashCode(callSuper = true)
public class SubmitChangeDTO extends BaseSubmitDTO {

	private String taskId;

	private TaskAction action;

	private Integer index;

}

File: src/dto/SubmitChangeDTO.ts
import { BaseSubmitDTO } from './BaseSubmitDTO';
import { TaskAction } from '../enums/TaskAction';

/**
 * Variation task submission parameters
 */
export class SubmitChangeDTO extends BaseSubmitDTO {
  /** Task ID */
  taskId?: string;

  /** UPSCALE(Enlarge); VARIATION(Variation); REROLL(Regenerate) */
  action?: TaskAction;

  /** Index (1~4), required when action is UPSCALE or VARIATION */
  index?: number;
}

